synfig-studio/src/gui/trees/layertreestore.cpp

```diff
@@ -1129,8 +1129,14 @@ LayerTreeStore::on_layer_raised(synfig::Layer::Handle layer)
 void
 LayerTreeStore::on_layer_moved(synfig::Layer::Handle layer,int depth, synfig::Canvas::Handle /*canvas*/)
 {
+	//if the layer was selected before movement it should remain selected
+	std::list<etl::handle<Layer>> selected_layer_list  = get_canvas_interface()->get_selection_manager()->get_selected_layers();
 	on_layer_removed(layer);
 	on_layer_inserted(layer,depth);
+	if (!selected_layer_list.empty()){
+		canvas_interface()->get_selection_manager()->clear_selected_layers();
+		canvas_interface()->get_selection_manager()->set_selected_layers(selected_layer_list);
+	}
```

Comment:
maybe not in this PR? Or is it important to this PR itself?

Reply:
Yes it's important to this PR so that when the layer is moved all the way to the top/bottom it remains selected.